### DISCUSSION TITLE
Error: Cannot find module '.next\server\pages-manifest.json'

### DISCUSSION THREAD

**Discussion by sergiuchuckmisha** [vercel/next.js#36194] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: #86~18.04.1-Ubuntu SMP Fri Jun 18 01:23:22 UTC 2021
    Binaries:
      Node: 16.14.2
      npm: 8.5.0
      Yarn: 1.22.5
      pnpm: N/A
    Relevant packages:
      next: 12.1.5
      react: 18.0.0
      react-dom: 18.0.0

also

    Platform: linux
      Arch: x64
      Version: #86~18.04.1-Ubuntu SMP Fri Jun 18 01:23:22 UTC 2021
    Binaries:
      Node: 16.14.2
      npm: 8.5.0
      Yarn: 1.22.5
      pnpm: N/A
    Relevant packages:
      next: 12.1.6-canary.1
      react: 18.0.0
      react-dom: 18.0.0

### Describe the Bug

actual result:
`/usr/bin/yarn run build
yarn run v1.22.5
$ next build
info  - Checking validity of types  
info  - Creating an optimized production build  
info  - Compiled successfully

> Build error occurred
[Error: ENOENT: no such file or directory, open '/home/misha/IdeaProjects/tmp/next-hw/.next/server/pages-manifest.json'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/misha/IdeaProjects/tmp/next-hw/.next/server/pages-manifest.json'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
`

next.config.js
`/** @type {import('next').NextConfig} */
const nextConfig = {
  webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
    // Important: return the modified config
    Object.assign(config, {output: {
        globalObject: 'this'
      }})
    return config
  },
  reactStrictMode: true,
}

module.exports = nextConfig
`

 - motivation: want to add [jsoneditor](https://www.npmjs.com/package/jsoneditor) to my next project; and, according to [this](https://stackoverflow.com/a/64639975/6045233) I need `globalObject: 'this',` for webpack, which is reflected in provided `next.config.js`
 
 after adding this option `globalObject: 'this',` appears `[Error: ENOENT: no such file or directory, open '/home/misha/IdeaProjects/tmp/next-hw/.next/server/pages-manifest.json']` 

### Expected Behavior

no errors after `npm run build`

### To Reproduce

- create new project with `npx create-next-app`
- modify `next.config.js` like this:
`/** @type {import('next').NextConfig} */
const nextConfig = {
  webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
    // Important: return the modified confi
    Object.assign(config, {output: {
        globalObject: 'this'
      }})
    return config
  },
  reactStrictMode: true,
}

module.exports = nextConfig`
- `npm run build`

**Comment by james-elicx** [comment]
Hello,

The problem is that you were overwriting the entire `output` object, instead of just the `globalObject` value in it.

Having just encountered this issue, there are a couple of options to fix it.

1. Spread the `config.output` object inside your `Object.assign`

```js
const nextConfig = {
  webpack: (config) => {
    Object.assign(config, {
      output: {
        ...config.output,
        globalObject: `(typeof self !== 'undefined' ? self : this)`,
      },
    });

    return config;
  },
};
```

2. Instead of using `Object.assign`, use `config.output.globalObject = ...`

```js
const nextConfig = {
  webpack: (webpackConfig) => {
    config.output.globalObject = `(typeof self !== 'undefined' ? self : this)`;

    return config;
  },
};
```
